tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc

```diff
@@ -2348,6 +2346,38 @@ TEST_F(EnergyPlusFixture, TestAFNPressureStat)
     EXPECT_EQ(0.0, SurfaceWindow(5).VentingOpenFactorMultRep);
     EXPECT_EQ(0.0, SurfaceWindow(14).VentingOpenFactorMultRep);
 
+    // Test for #7162
+    DataHeatBalFanSys::ZoneAirHumRat.allocate(4);
+    DataHeatBalFanSys::MAT.allocate(4);
+    DataHeatBalFanSys::ZoneAirHumRatAvg.allocate(NumOfZones);
+
+    DataHeatBalFanSys::MAT(1) = 23.0;
+    DataHeatBalFanSys::MAT(2) = 23.0;
+    DataHeatBalFanSys::MAT(3) = 23.0;
+    DataHeatBalFanSys::MAT(4) = 5.0;
+    DataHeatBalFanSys::ZoneAirHumRat(1) = 0.0007;
+    DataHeatBalFanSys::ZoneAirHumRat(2) = 0.0011;
+    DataHeatBalFanSys::ZoneAirHumRat(3) = 0.0012;
+    DataHeatBalFanSys::ZoneAirHumRat(4) = 0.0008;
+    DataHeatBalFanSys::ZoneAirHumRatAvg = DataHeatBalFanSys::ZoneAirHumRat;
+    DataZoneEquipment::ZoneEquipConfig.allocate(4);
+    DataZoneEquipment::ZoneEquipConfig(1).IsControlled = false;
+    DataZoneEquipment::ZoneEquipConfig(2).IsControlled = false;
+    DataZoneEquipment::ZoneEquipConfig(3).IsControlled = false;
+    DataZoneEquipment::ZoneEquipConfig(4).IsControlled = false;
+    DataHVACGlobals::TimeStepSys = 0.1;
+
+    AirflowNetwork::AirflowNetworkLinkSimu(1).FLOW2 = 0.1;
+    AirflowNetwork::AirflowNetworkLinkSimu(10).FLOW2 = 0.15;
+    AirflowNetwork::AirflowNetworkLinkSimu(13).FLOW2 = 0.1;
+
+    ReportAirflowNetwork();
+
+    EXPECT_NEAR(34.3673036, AirflowNetwork::AirflowNetworkReportData(1).MultiZoneInfiLatGainW, 0.0001);
+    EXPECT_NEAR(36.7133377, AirflowNetwork::AirflowNetworkReportData(2).MultiZoneMixLatGainW, 0.0001);
+    EXPECT_NEAR(89.3450925, AirflowNetwork::AirflowNetworkReportData(3).MultiZoneInfiLatLossW, 0.0001);
+
+    AirflowNetwork::AirflowNetworkExchangeData.deallocate();
```

Comment:
Array deallocation at end of unit tests is not standard. All deallocations should be performed by clear_state functions.

Reply:
@rraustad The array is deallocated in clear_state() of Elements.cpp. I am going to remove it after resolving conflict.